pyaerocom/aeroval/_processing_base.py

```diff
@@ -69,6 +73,222 @@ def run(self, *args, **kwargs) -> list:
         """
         pass
 
+    def _add_heatmap_timeseries_entry(
```

Comment:
The main reason was to avoid having to either pass in an aerovaldb instance to the function, or open a new connection in each function call, neither of which I particularly like. How would you feel about moving these to `ExperimentOutput`? As far as I can see each Engine class has an instance of ExperimentOutput, so self.experiment_output._add_heatmap... would work.